ocs-api/import-owner-private-keys2.sh

```diff
@@ -0,0 +1,300 @@
+#!/bin/bash
+# This script will extract a tarfile containing all private keys and certs necessary to add the Key-Pairs to our master keystore inside the container for Owner Attestation.
+
+if [[ "$1" == "-h" || "$1" == "--help" ]]; then
+    cat << EndOfMessage
+Usage: ${0##*/} <org-id> <key-name> <common-name> <email-name> <company-name> <country-name> <state-name> <locale-name>
+
+Arguments:
+  <org-id> - The organizational unit or Horizon Org ID the user is in. Necessary information for keyCertificate generation.
+  <key_name> - The custom key pair name the user chooses. Necessary information to keep track of imported keystores into our master keystore.
+  <common-name> - The name of the user. Necessary information for keyCertificate generation.
+  <email-name> - The user's email. Necessary information for keyCertificate generation.
+  <company-name> - The company the user works for. Necessary information for keyCertificate generation.
+  <country-name> - The country the user resides in. Necessary information for keyCertificate generation.
+  <state-name> - The state the user resides in. Necessary information for keyCertificate generation.
+  <locale-name> - The city the user resides in. Necessary information for keyCertificate generation.
+
+EndOfMessage
+    exit 0
+fi
+
+#if [[ -z "$1" ]]; then
+#  echo "Error: Certificate arguments not specified"
+#    exit 1
+#fi
+
+#Make all positional arguments required. Try to check number of arguments passed? $# -lt 7
+if [[ -z "$1" ]] || [[ -z "$2" ]] || [[ -z "$3" ]] || [[ -z "$4" ]] || [[ -z "$5" ]] || [[ -z "$6" ]] || [[ -z "$7" ]] || [[ -z "$8" ]]; then
+    echo "Error: All json values were not specified" >&2
+    exit 1
+fi
+
+
+#Grabbing password from the ocs/config/application.properties inside the container to use for import
+keypwd="$(grep -E '^ *FS_OWNER_KEYSTORE_PASSWORD=' ocs/ocs.env)"
+SDO_KEY_PWD=${keypwd#FS_OWNER_KEYSTORE_PASSWORD=}
+keyType="all"
+gencmd=""
+str="$(openssl version)"
+
+#Positional Arguments for ocs api
+HZN_ORG_ID="$1"
+ORG_UNIT=$HZN_ORG_ID
+KEY_NAME="${2:-Default}"
+KEY_NAME=$(echo "$KEY_NAME" | tr '[:upper:]' '[:lower:]')
+COMMON_NAME="${3:-Default}"
+EMAIL_NAME="${4:-default@gmail.com}"
+COMPANY_NAME="${5:-Default}"
+COUNTRY_NAME="${6:-US}"
+STATE_NAME="${7:-North Carolina}"
+LOCALE_NAME="${8:-Raleigh}"
```

Comment:
I force the 'KEY_NAME' to lowercase because that is also what the master keystore does once the alias is imported. Every alias' inside the master keystore is in lowercase. I figured since the alias inside the master keystore was a forced lowercase, then the public key file should be all lowercase as well for consistency purposes

Reply:
I see. That is good to keep the private key and public key names the same. I'm still concerned that the user won't know we did this. For example:
1. User runs your post api and gives you a key name of `MyNewKey`
2. You create the private key and public key with key name `mynewkey`
3. You return the bytes of the public key and they direct that to filename `MyNewKey.pub` (because they think that is what they named it)
4. They run the `GET /keys` api and the listing includes a public key named `mynewkey.pub` and they wonder if that is their key or someone else's

Once the `hzn sdo key create` command is implemented, it will help this situation, because it can store the key file in their local dir and name it `mynewkey.pub`. Maybe that will be good enough? Or maybe we should document that the key name needs to be all lowercase, and reject the POST if it's not?